File: backend/src/main/java/com/mikelike/getpethelp/backend/domain/FullUserInfo.java
package com.mikelike.getpethelp.backend.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="full_user_info")
public class FullUserInfo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToOne
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    User user;
    private String middleName;
    private int age;
    private String city;
    private boolean telegram;
    private boolean whatsApp;
    private boolean viber;
}

File: mobileClient/app/src/main/java/com/mikelike/getpethelp/mobile/domain/FullUserInfo.kt
package com.mikelike.getpethelp.mobile.domain

data class FullUserInfo(
    val id: Long,
    val user: User,
    val middleName: String,
    val age: Int,
    val city: String,
    val telegram: Boolean,
    val whatsApp: Boolean,
    val viber: Boolean
)